### PR TITLE
Mapl collection files

### PR DESCRIPTION
**Pull request by marcopluess**

### CODE REVIEW

**Comment by marcopluess** [comment]
tested with following markup in cam:

```haml
.mtl-layout-default-content
  .container
    .row
      .col.s12
        .collection-files
          %a.card-panel{ href:'/path/to/file.pdf', title:'Vertrag Dolorem.jpg' }
            Vertrag Dolorem
            %span
              %i.material-icons.red-text picture_as_pdf
              PDF
            %i.close.material-icons close

          %a.card-panel{ href: '...', title:'file.xlsx' }
            %strong Loss of I don't know really what, wtf is this?
            file.xlsx
            %span.grey-text
              %i.material-icons.blue-text insert_drive_file
              Bild
            %i.close.material-icons close

          %a.card-panel.no-padding{ href:'...', title: 'Loss of i don\'t know (didum.jpg)' }
            %img{ alt:'Loss of i don\'t know (didum.jpg)', src:'http://i.giphy.com/l2JJsnBRsvr9bs5RC.gif' }
            %i.close.material-icons close
```

**Comment by marcopluess** [comment]
@lwe: want to proceed with the integration of the actual helper?

(https://gist.github.com/marcopluess/a127badf9b9f820d524c251dfc44abd0)

**Comment by lwe** [comment]
👌, will start Monday. Thanks for the markup + styling.

File: app/assets/stylesheets/mtl/extend/_collection-files.scss [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    width: calc(50% - 20px);
+    min-width: $mtl-collection-file-minwidth;
+    min-height: $mtl-collection-file-minheight;
+    position: relative;
+    overflow: visible;
+    font-size: 13px;
+    line-height: 16px;
+
+    &:hover {
+      box-shadow: 0 4px 8px rgba(color('shades', 'black'), .26);
```
Comment (lwe) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: There is `.hoverable`, maybe we can simply @extend / @include it maybe?

File: app/assets/stylesheets/mtl/extend/_collection-files.scss [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      .material-icons {
+        line-height: 16px;
+        height: 16px;
+        font-size: 20px;
+        vertical-align: middle;
+      }
+    }
+
+    // image
+    img {
```
Comment (lwe) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: how does this work with image scaling, e.g. if the image has a different aspect ratio than our container?

File: app/assets/stylesheets/mtl/extend/_collection-files.scss (hunk 1 above)
Comment (marcopluess) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: hoverable would kill the panels default boxshadow. I'd prefer to keep it plain like this.

**Comment by lwe** [comment]
Uhm, one more thing...

```haml
%a.card-panel.no-padding
  %img
```

The `.no-padding` is not an option sorry, can we work with negative margins or something?

File: app/assets/stylesheets/mtl/extend/_collection-files.scss [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  margin: (-$mtl-collection-files-spacing) 0 0 (-$mtl-collection-files-spacing);
+  @include clearafter;
+
+  // extending the card panel
+  .card-panel {
+    background-color: $mtl-collection-file-bg;
+    color: $mtl-collection-file-text-primary;
+    margin: $mtl-collection-files-spacing 0 0 $mtl-collection-files-spacing;
+    float: left;
+    width: calc(50% - 20px);
```
Comment (lwe) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: are we sure we want to use calc, also this only works for to column layouts right?

File: app/assets/stylesheets/mtl/extend/_collection-files.scss [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    min-width: $mtl-collection-file-minwidth;
+    min-height: $mtl-collection-file-minheight;
+    position: relative;
+    overflow: visible;
```
Comment (lwe) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: shouldn't this be `hidden`, or is this for the close button?

File: app/assets/stylesheets/mtl/extend/_collection-files.scss [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    font-size: 13px;
```
Comment (lwe) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: 1rem

File: app/assets/stylesheets/mtl/extend/_collection-files.scss (hunk 2 above)
Comment (marcopluess) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: we have to fine-tune this later... check the comment I'm about to add below in a minute.

File: app/assets/stylesheets/mtl/extend/_collection-files.scss [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    line-height: 16px;
```
Comment (lwe) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: no pixel, use e.g. 1.3 or whatever the default line-height is - or even reference a variable

File: app/assets/stylesheets/mtl/extend/_collection-files.scss [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+    &:hover {
+      box-shadow: 0 4px 8px rgba(color('shades', 'black'), .26);
+    }
+
+    > strong,
+    > span {
+      display: block;
+      font-size: 13px;
```
Comment (lwe) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: neither `font-size` nor `line-height` should be required, so I propose the remove

File: app/assets/stylesheets/mtl/extend/_collection-files.scss [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    }
+    > span {
+      color: $mtl-collection-file-text-secondary;
+      vertical-align: middle;
+
+      // icon override 1
+      .material-icons {
+        line-height: 16px;
+        height: 16px;
+        font-size: 20px;
```
Comment (lwe) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: use rem's

**Comment by marcopluess** [comment]
noup, not really possible. we could add it as a background image and use `background-size: cover;`. this would fix the ratio issues, on the cost of risking a highly scaled image if it has a portrait format.

File: app/assets/stylesheets/mtl/extend/_collection-files.scss [hunk 9]
```diff
@@ -0,0 +1,91 @@
+// variables / colors
+$mtl-collection-files-spacing: 20px;
```
Comment (lwe) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: use rem's

File: app/assets/stylesheets/mtl/extend/_collection-files.scss (hunk 3 above)
Comment (marcopluess) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: yeah, the sizes have to be discussed still. flo proposed a two column layout on a flexible container, thus 50%. that could not be matched with a fixed size. TBD

File: app/assets/stylesheets/mtl/extend/_collection-files.scss (hunk 4 above)
Comment (marcopluess) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: yes, close button!

**Comment by lwe** [comment]
Maybe use `.card-panel.card-panel-image` instead?

File: app/assets/stylesheets/mtl/extend/_collection-files.scss (hunk 5 above)
Comment (marcopluess) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: 1rem would be about 14.5px. will add 0.897rem to match the styling.

**Comment by marcopluess** [comment]
if we go down the rem road, I'll add a helper mixin for maintainability.

File: app/assets/stylesheets/mtl/extend/_collection-files.scss (hunk 7 above)
Comment (marcopluess) [inline app/assets/stylesheets/mtl/extend/_collection-files.scss]: ? the default font size is too big even for medium font-sizes. or should I just plainly ignore the sizes proposed in flo's sketch?

**Comment by marcopluess** [comment]
and why is `.card-panel-image` better than the normalized approach of `.no-padding`?

you can even know what it does by reading the class name.

**Comment by lwe** [comment]
1. Re: rem's, at least for font-sizes, line-height I'm very sure we should use rem (or for line-height, the relative notation, e.g. 1.3), so maybe we should just do that atm
2. Re: `.card-panel-image` was toying with the idea to use the following markup (see below) and then simply hide the inner content, except the img, so the helper would always generate the same markup...

```haml
%a.card-panel.card-panel-image
  %img src: '...'
  %strong Didum
  ...

**Comment by marcopluess** [comment]
REM's: I'd have to use a converter to convert the px's to rem's:

```sass
// px to rem converter
@function rem($num) {
  @return ($num / 14.5) + rem;
}

.element {
  font-size: rem(13);
}
```

We had such a converter in bautro and ended up not using it. MaterializeCSS also uses px-values for spacings and such, better to say: The have already a mix and match between PX and REM values, so no matter what we do, it won't be perfect... but it can surely give me a hard time. See examples from their own styleguide:

screen shot 2016-08-05 at 17 31 37

screen shot 2016-08-05 at 17 31 56

**Comment by lwe** [comment]
Anyway, after some discussions:

- We use whatever we want as font-size and sizes, ignore the rem comments then
- `no-padding` is fine too IMO, after revising the scenario and options

**Comment by marcopluess** [comment]
@lwe: should be good now... please review and merge.

**Comment by lwe** [comment]
Thanks, merged 👍

**Pull request merged**